Dockerfile

```diff
@@ -16,8 +16,8 @@ RUN useradd c-api-user && mkdir -p /home/c-api-user && chown c-api-user /home/c-
 ENV HOME /home/c-api-user
 ENV LD_LIBRARY_PATH /usr/lib/llvm-3.8/lib:$LD_LIBRARY_PATH
 
-USER c-api-user
-RUN git clone --recurse-submodules https://github.com/jbcoe/C_API_generation /home/c-api-user/demo
+COPY . /home/c-api-user/demo
+RUN cd /home/c-api-user/demo && git submodule update --init --recursive
```

Comment:
The new workflow is below:
The docker worker receives the PR branch, checks it out and starts building the docker container. 

During docker build:
The docker daemon copies the working tree and initialises submodules. If you look at the [documentation of git-clone](https://git-scm.com/docs/git-clone#git-clone---recurse-submodules) recurse-submodules it’s equivalent to git submodule update --init –recursive after cloning. We replace clone with copy and have to manually init all submodules. I tried to git-clone from localhost onto the container, but the only way to do this now is to mount a volume and that's only possible during docker run, so copying seemed to work the best.

So we fixed 2 problems:
Guarantee that travis checks out the correct branch before starting docker build
Docker build doesn’t clone the repo, but instead uses the current working tree

This approach allows you to run tests locally and they should be the same as on Travis-CI